app/workers/vendor_api_request_worker.rb

```diff
@@ -0,0 +1,37 @@
+class VendorAPIRequestWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 3, queue: :low_priority
+
+  def perform(request_data, response_body, status_code, created_at)
+    request_headers = request_data['headers']
+    provider_id = provider_id_from_auth_token(request_headers.delete('HTTP_AUTHORIZATION'))
+
+    VendorAPIRequest.create!(
+      request_path: request_data['path'],
+      request_headers: request_headers,
+      request_body: request_data['params'],
+      response_body: response_hash(response_body, status_code),
+      status_code: status_code,
+      provider_id: provider_id,
+      created_at: created_at,
+    )
+  end
+
+private
+
+  def provider_id_from_auth_token(auth_header)
+    return if auth_header.blank?
+
+    token = auth_header[7..-1]
```

Comment:
Not sure, but perhaps we can use the same mechanism as the controller to extract the token?

See https://github.com/DFE-Digital/apply-for-teacher-training/commit/36c71e86d37e4e4fbffee969ee839edc4645edf3

Otherwise it's possible that the middlware doesn't recognise tokens that the controller does - I think Rails' `authenticate_with_http_token` is more liberal in what it accepts.

Reply:
Feel free to ignore if it doesn't work.

Reply:
Thanks @tijmenb yeah it seems prudent to be consistent with the controller. 
`authenticate_with_http_token` calls an underlying method `ActionController::HttpAuthentication::Token#token_and_options` https://github.com/rails/rails/blob/070d4afacd3e9721b7e3a4634e4d026b5fa2c32c/actionpack/lib/action_controller/metal/http_authentication.rb#L457 which seems like a better fit for a Sidekiq worker. Added here https://github.com/DFE-Digital/apply-for-teacher-training/pull/3009/commits/da01857080650c9abf1027a6d78fb7f2e805c7c5#diff-ea82dc5cde409c49f45d4efc687841c9R29